pkg/controller/certificates/keymanager/keymanager_controller.go

```diff
@@ -154,6 +155,10 @@ func (c *controller) ProcessItem(ctx context.Context, key types.NamespacedName)
 		return nil
 	}
 
+	// Apply runtime defaults to apply default values that are governed by
+	// controller feature gates, such as DefaultPrivateKeyRotationPolicyAlways.
+	cminternal.SetRuntimeDefaults_Certificate(crt)
```

Comment:
For runtime defaults, we probably want to create a read-only copy that we use as input for the follow-up steps.
By mutating the original object, we risk updating the resource when we "update"?

Reply:
I thought more about this. I shouldn't be mutating the runtime object without first doing a deep-copy.

I'll create a followup PR.